src/tribler/gui/tribler_window.py

```diff
@@ -395,6 +383,39 @@ def sigint_handler(*_):
             upgrade_manager=self.upgrade_manager,
         )
 
+    def restore_window_geometry(self):
+        screen_geometry: QRect = QApplication.desktop().availableGeometry()
+        size: QSize = self.gui_settings.value("size", self.size())
+
+        def restore_size():
+            self._logger.info(f'Available screen geometry: {screen_geometry}')
+            self._logger.info(f'Restored window size: {size}')
+
+            bounded_size = QSize(
+                min(size.width(), screen_geometry.width()),
+                min(size.height(), screen_geometry.height())
+            )
+            self._logger.info(f'Resize window to the bounded size: {bounded_size}')
+            self.resize(bounded_size)
+
+        def restore_position():
+            pos = self.gui_settings.value("pos", self.pos())
+            self._logger.info(f'Restored window position: {pos}')
+
+            window_geometry = QRect(pos, size)
+            union: QRect = screen_geometry | window_geometry
+            delta_x = union.width() - screen_geometry.width()
+            delta_y = union.height() - screen_geometry.height()
+            actual_position = QPoint(
+                pos.x() + delta_x,
+                pos.y() + delta_y
```

Comment:
This logic is incorrect. `delta_x` and `delta_y` are always non-negative, as the union is always bigger than `screen_geometry`.
When `delta_x` and `delta_y` are added to `pos`, `pos` moves right and down and never left and up. If the right and bottom part of the Tribler window is placed out of the screen, each time the Tribler is restarted, the Tribler windows will move further and further right and down until it is entirely out of sight.

Reply:
Thank you, you are right. 

I've removed the last commit.